### PR TITLE
Improve error handling around crm_mon's daemonize mode

### PR DESCRIPTION
**Pull request by clumens**
The basic problem here is that if you run `crm_mon -d --output-to foo`, it creates a foo file and puts nothing in it because the output format is none.  Tell the user they need to specify an output format.

### CODE REVIEW

**Review by kgaillot — COMMENTED** [review]
If we keep going like this, crm_mon will actually start to make sense ;)

File: tools/crm_mon.c [hunk 1]
```diff
@@ -1550,7 +1531,31 @@ main(int argc, char **argv)
         return clean_up(CRM_EX_ERROR);
     }
 
-    /* output_format MUST NOT BE CHANGED AFTER THIS POINT. */
+    if (options.daemonize) {
+        if (output_format == mon_output_console || output_format == mon_output_unset || output_format == mon_output_none) {
+            g_set_error(&error, PCMK__EXITC_ERROR, CRM_EX_USAGE,
+                        "--daemonize requires --output-as=[html|text|xml]");
```
[the remark this answers is not in the document]
Comment (clumens) [reply tools/crm_mon.c:1537]: What would be the point of that?

File: tools/crm_mon.c [hunk 2]
```diff
@@ -1385,10 +1385,18 @@ one_shot(void)
     clean_up(CRM_EX_OK);
 }
 
+static crm_exit_t
+invalid_cib(void)
+{
+    // Shouldn't really be possible
+    g_set_error(&error, PCMK__EXITC_ERROR, CRM_EX_ERROR, "Invalid CIB source");
+    return clean_up(CRM_EX_ERROR);
```
[the remark this answers is not in the document]
Comment (clumens) [reply tools/crm_mon.c:1393]: Fixed.

File: tools/crm_mon.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (clumens) [reply tools/crm_mon.c:1537]: Okay, I see.

File: tools/crm_mon.c [hunk 3]
```diff
@@ -1494,8 +1494,10 @@ main(int argc, char **argv)
             }
 
         } else if (options.daemonize) {
-            if (pcmk__str_eq(args->output_dest, "-", pcmk__str_null_matches | pcmk__str_casei) && !options.external_agent) {
-                g_set_error(&error, PCMK__EXITC_ERROR, CRM_EX_USAGE, "--daemonize requires at least one of --output-to and --external-agent");
+            if (pcmk__str_eq(args->output_dest, "-", pcmk__str_null_matches|pcmk__str_casei) &&
```
Comment (kgaillot) [inline tools/crm_mon.c:1497]: pcmk__str_casei isn't needed for "-" :)

**Review by kgaillot — COMMENTED** [review]
looks good to me

**Pull request merged**